### ISSUE TITLE
The code execution cannot proceed because [x].dll was not found.

### ISSUE THREAD

**Issue by MarvinG6547** [Trico-Everfire/p2-publishing-tool#20]
Every .dll file is named wrong.
The program expects all of them to have the letter d at the end of their name.
For example: Zip includes qt6core.dll but the program expects qt6cored.dll.
As stated before every file is like this.
It even expects files that aren't included with the program to have this naming convention, and after renaming all the files to fit it's needs it says that it can't find the exact same files but that don't have the letter d (ie. the files before they were renamed).
After doing all of that it then spits out the error in the below attached image:
![bug](https://user-images.githubusercontent.com/117035792/219901181-a9bd1c9a-512c-4b19-8f89-bdde5349ff7f.JPG)

**Comment by Trico-Everfire** [comment]
You probably downloaded the "Debug" build, although provided, I do not ship QT's debug libraries, please download the Release.

**Comment by Trico-Everfire** [comment]
Currently, this build is also missing SSL libraries, I'll be providing a new build around 2 minutes after this message is posted. Remember to download the Release, thank you.

**Comment by MarvinG6547** [comment]
Well I did also download the release but then the program says that it couldn't start any qt platform
![eh](https://user-images.githubusercontent.com/117035792/219905171-8f13018b-7f9d-4a8b-8a38-36772c24a923.JPG)

**Comment by Trico-Everfire** [comment]
I think I am aware of what's causing it, should be fixed in a short moment.

**Comment by Trico-Everfire** [comment]
missing qminimal has been fixed in newest release.

**Comment by Trico-Everfire** [comment]
If there are no futher issues, please notify me so I can close this issue.

**Comment by MarvinG6547** [comment]
Looks like there is a new issue
![eh](https://user-images.githubusercontent.com/117035792/219946266-5a6dd468-70a9-44fd-aecf-e78705a93132.JPG)

**Comment by MarvinG6547** [comment]
Ok so I reverted the files that I added the letter d to back to their original names and now the error actually managed to load.
![huh](https://user-images.githubusercontent.com/117035792/219946678-ad4a6c53-aed9-44c3-a6b0-e4947371865c.JPG)

**Comment by jple777** [comment]
Same issue here with same screenshots

**Comment by Trico-Everfire** [comment]
I am still missing QT some libraries, QT is an amazing UI framework, but doesn't end up telling you what you exactly need to pack when it comes to releasing your software. that's mostly why this is a beta, because of missing dependencies. I'll have a build ready in a few minutes, hopefully with the last of the missing stuff.

**Comment by Trico-Everfire** [comment]
it may take longer, it's a bit more complex. Apparently requiring a different library and linking it to the application.

**Comment by Trico-Everfire** [comment]
[p2-publishing-tool-windows-Release.zip](https://github.com/Trico-Everfire/p2-publishing-tool/files/10777105/p2-publishing-tool-windows-Release.zip)

Making another release for another test seems silly, so here's the download directly.
@MarvinG6547 
@jple777

**Comment by Trico-Everfire** [comment]
if it's the same "missing tls" issue, try moving the dll from the tls folder to where the exe is.

**Comment by MarvinG6547** [comment]
There is no tls folder

**Comment by Trico-Everfire** [comment]
:facepalm: I forgot about the artifacts! hold on.

**Comment by Trico-Everfire** [comment]
[p2-publishing-tool-windows-Release.zip](https://github.com/Trico-Everfire/p2-publishing-tool/files/10777248/p2-publishing-tool-windows-Release.zip)

Here we go, should do it.

**Comment by MarvinG6547** [comment]
You should really fix that picture
![ijrohg](https://user-images.githubusercontent.com/117035792/219960761-35f770bd-60b5-40ae-a9c0-aa26a6267f45.JPG)

**Comment by Trico-Everfire** [comment]
I actually don't know why the error, Failed to retrieve image is QT's innability to retrieve the image from the URL. can you send me the URL real quick?

**Comment by MarvinG6547** [comment]
The link just leads to a 404
https://steamuserimages-a.akamaihd.net/ugc/%202029469358271870432/961FB37C552088C2E96A2052085E5845C65E0452/

**Comment by Trico-Everfire** [comment]
> You should really fix that image
That image is one of the URL images Steam has of your items. it's how the "edit" thumbnail preview works. it's unable to read this image, this could be a firewall issue, as QT doesn't seem to stress  in the console, which means the request went through, just didn't ass/

**Comment by MarvinG6547** [comment]
Well it's not being blocked by my firewall
![lkwjeng](https://user-images.githubusercontent.com/117035792/219961391-ed4b0275-b7f4-484e-af2c-f288fbee6725.JPG)

**Comment by Trico-Everfire** [comment]
> The link just leads to a 404 https://steamuserimages-a.akamaihd.net/ugc/%202029469358271870432/961FB37C552088C2E96A2052085E5845C65E0452/

Yeah, that should do it, idk why though, it's a link directly fetched from Steam itself, I don't have control over that, I'll have to look into that, because it's either a Steam problem, or something else entirely.

**Comment by MarvinG6547** [comment]
I mistyped the link it isn't a 404
It's the thumbnail to my latest map?

**Comment by MarvinG6547** [comment]
It's just an in-game screenshot so I wouldn't expect it to be corrupt

**Comment by Trico-Everfire** [comment]
Correct, it  is the thumbnail to your latest map. This is because it needs it for the image display when you edit a workshop item. if it isn't because of missing data, It has to be something else. I'll look into it.

**Comment by MarvinG6547** [comment]
I do have this file in my screenshots folder that mentions akamai, which is the same webpage in the screenshot link but I highly doubt that it's of any use
![wger](https://user-images.githubusercontent.com/117035792/219962584-a277d737-3204-483a-903d-c50711957ea7.JPG)

**Comment by Trico-Everfire** [comment]
[p2-publishing-tool-windows-Release.zip](https://github.com/Trico-Everfire/p2-publishing-tool/files/10777416/p2-publishing-tool-windows-Release.zip)
This shouldn't solve the problem, but rather provide you with an error code when it fails, I need that error code if I wanna know how to fix this.

**Comment by MarvinG6547** [comment]
![wef](https://user-images.githubusercontent.com/117035792/219962716-e24618dc-ca47-4a4a-821c-5a851ad20e91.JPG)
Interesting

**Comment by Trico-Everfire** [comment]
> I do have this file in my screenshots folder that mentions akamai, which is the same webpage in the screenshot link but I highly doubt that it's of any use ![wger](https://user-images.githubusercontent.com/117035792/219962584-a277d737-3204-483a-903d-c50711957ea7.JPG)

afaik, akamai is a hosting service, the link with your thumbnail is directly retrieved from Steam through the Steamworks API, this link is then processed by QT to download the thumbnail image.

**Comment by Trico-Everfire** [comment]
> ![wef](https://user-images.githubusercontent.com/117035792/219962716-e24618dc-ca47-4a4a-821c-5a851ad20e91.JPG) Interesting

Then according to my findings, it... fails to read the image, it appears to retrieve it, just fails to read it. why? I don't know.

**Comment by MarvinG6547** [comment]
Could it be that it doesn't like .jpg files?

**Comment by Trico-Everfire** [comment]
Okay, one more error reading.
[p2-publishing-tool-windows-Release.zip](https://github.com/Trico-Everfire/p2-publishing-tool/files/10777446/p2-publishing-tool-windows-Release.zip)

**Comment by MarvinG6547** [comment]
![gr](https://user-images.githubusercontent.com/117035792/219963366-5584bb51-fc7f-4825-aa8f-07f69ceb9239.JPG)

**Comment by Trico-Everfire** [comment]
> Could it be that it doesn't like .jpg files?

It actually really likes .jpg files. if you upload an image of an image above 1MB, it'll actually atuomatically convert the image to a jpg to downgrade the quality to fit 1MB (this is a limitation placed by STEAM and I wish I could change it but I don't own Valve Software.)

**Comment by Trico-Everfire** [comment]
> ![gr](https://user-images.githubusercontent.com/117035792/219963366-5584bb51-fc7f-4825-aa8f-07f69ceb9239.JPG)

Then according to the readouts, there shouldn't be an error :/
I am so confused right now.

**Comment by MarvinG6547** [comment]
Quick question are you intentionally ignoring that "Plugin uses incompatible Qt library" error?

**Comment by Trico-Everfire** [comment]
That is a missing debug dll error, I don't know why it's asking for a debug DLL, because this isn't a debug build.

**Comment by Trico-Everfire** [comment]
Are you still using the debug dlls from before?

**Comment by Trico-Everfire** [comment]
delete qwindowsd (only the d version).

**Comment by MarvinG6547** [comment]
That removed the error message but it still crashes
![heof](https://user-images.githubusercontent.com/117035792/219963874-86716dd0-82ce-48d4-837e-3589289d670e.JPG)

**Comment by Trico-Everfire** [comment]
What confuses me is the empty text box, it's probably because it's white on white, I need to fix the styling when I get to it. As to why, I'll look into it.

**Comment by MarvinG6547** [comment]
> You should really fix that image ![ijrohg](https://user-images.githubusercontent.com/117035792/219960761-35f770bd-60b5-40ae-a9c0-aa26a6267f45.JPG)

Well it did appear correctly earlier

**Comment by Trico-Everfire** [comment]
I've always had trouble with QT and Windows, this isn't my first rodeo, but it always seems to not wanna work and it doesn't tell me why. I'll contact some friends a bit more well versed and hopefully fix this for good.

**Comment by DrPeeper228** [comment]
maybe space character breaks it?

**Comment by Trico-Everfire** [comment]
> maybe space character breaks it?

It's weird then that it doesn't break on Linux, I know things are different between OSses, but QT is supposed to behave 1:1 between them, so I doubt it's anything with characters.

**Comment by Trico-Everfire** [comment]
this should fix the crashing regarding the qmessagebox warnings.
[p2-publishing-tool-windows-Release.zip](https://github.com/Trico-Everfire/p2-publishing-tool/files/10816825/p2-publishing-tool-windows-Release.1.zip)

**Comment by Trico-Everfire** [comment]
As to why exactly the image reader does not accept the network reply, I'll have to do some more research.

**Comment by Trico-Everfire** [comment]
https://github.com/Trico-Everfire/p2-publishing-tool/releases/tag/2.0.3-beta

**Comment by Trico-Everfire** [comment]
> https://github.com/Trico-Everfire/p2-publishing-tool/releases/tag/v2.0.4-beta

@MarvinG6547 
@jple777 

This should be the thing to fix this issue. If no items show up, let me know.

**Comment by Trico-Everfire** [comment]
Someone confirmed that the tool is functional to a degree, only issue is with the uploading of BSPs, but this specific issue has been deemed closed. May reopen if it ever comes up again.
